Module - mbed-drivers dep update to v1

0xc0170 opened the pull request:
Fixes "error: mbed-drivers does not meet specification ~0.12.0 required by core-util" reported by yotta for apps.

mbed-drivers currently depend on this module as ^1.0.0

@bogdanm @bremoran

bremoran commented:
This should be published as a major version increase. We need to break the dependency of core-util on mbed-drivers.

bremoran commented:
Actually, I've changed my mind. The only dependency of core-util on mbed-drivers is mbed-die. This API rarely changes, so it makes more sense to change this to a less specific match. I think this would be appropriate:

```JSON
    "mbed-drivers" : ">0.12.0"
```

0xc0170 commented:
Rebased, I added this to the commit message:

> As core-util for mbed requires mbed_die, which has not changed between the last version
used here and the current v1.0.0 in mbed drivers, using >=0.12.0,<2.0.0 should be safe.
Breaking dependency on mbed_die will require a breaking change in mbed drivers and in this module.

bremoran commented:
LGTM
